### PR TITLE
Reflow "Installation on Debian-based systems" for QGis server 

### PR DESCRIPTION
**Pull request by benoitdm-oslandia**
This PR is a part of [qgis server documentation enhancement](https://github.com/qgis/QGIS-Enhancement-Proposals/issues/184).

I reflowed the QGis server "Installation on Debian-based systems" documentation, including: 

* remove deprecated multiwatch
* fix redondant configuration parts
* change section depths to increase web server differences
* homogenize systemclt/service call
* homogenize rst code block usage
* add sample service unit for spawn-fcgi

No backport needed!
- [ ] Backport to LTR documentation is required

### CODE REVIEW

**Comment by havatv** [comment]
> I don't plan to backport this to current LTR

Why not? This should be relevant also for 3.10, or?

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 1]
```diff
@@ -34,56 +34,55 @@ Once the chosen repository is configured, installation is simply done with:
...
+     <ServiceException code="Service configuration error">Service unknown or unsupported</ServiceException>
+    </ServiceExceptionReport>
 
-Let's add a sample project. You can use your own, or one from
-`Training demo data <https://github.com/qgis/QGIS-Training-Data/>`_:
+  Let's add a sample project. You can use your own, or one from
+  `Training demo data <https://github.com/qgis/QGIS-Training-Data/>`_:
 
-.. code-block:: bash
+  .. code-block:: bash
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:63]:
```suggestion
.. code-block:: bash
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 2]
```diff
@@ -34,56 +34,55 @@ Once the chosen repository is configured, installation is simply done with:
...
+  
+    mkdir /home/qgis/projects/
+    cd /home/qgis/projects/
+    wget https://github.com/qgis/QGIS-Training-Data/archive/v2.0.zip
+    unzip v2.0.zip
+    mv QGIS-Training-Data-2.0/exercise_data/qgis-server-tutorial-data/world.qgs .
+    mv QGIS-Training-Data-2.0/exercise_data/qgis-server-tutorial-data/naturalearth.sqlite .
+  
+  Of course, you can use your favorite GIS software to open this file and
+  take a look on the configuration and available layers.
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:73]:
```suggestion
Of course, you can use your favorite GIS software to open this file and
take a look at the configuration and available layers.
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 3]
```diff
@@ -34,56 +34,55 @@ Once the chosen repository is configured, installation is simply done with:
...
 
- mkdir /home/qgis/projects/
- cd /home/qgis/projects/
- wget https://github.com/qgis/QGIS-Training-Data/archive/v2.0.zip
- unzip v2.0.zip
- mv QGIS-Training-Data-2.0/exercise_data/qgis-server-tutorial-data/world.qgs .
- mv QGIS-Training-Data-2.0/exercise_data/qgis-server-tutorial-data/naturalearth.sqlite .
+* To properly deploy QGIS server you need a HTTP server. Recommended choices are **Apache** or **NGINX**.
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:75]:
```suggestion
Deploying QGIS server
------------------------------------------------
To properly deploy QGIS server you need a HTTP server. Recommended choices are **Apache** or **NGINX**.
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 4]
```diff
@@ -34,56 +34,55 @@ Once the chosen repository is configured, installation is simply done with:
...
+.. index:: Apache, mod_fcgid
 
 .. _`httpserver`:
 
-HTTP Server configuration
--------------------------
-
-To run QGIS server you need a web server. Recommended choices are **Apache** or **Nginx**.
+Apache HTTP Server
+------------------
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:82]:
```suggestion
..................
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 5]
```diff
@@ -134,7 +133,7 @@ Let's now add ``mod_fcgid`` configuration directives for QGIS Server:
 .. note::
 
   See the ``mod_fcgid`` documentation for more information on the ``Fcgid`` parameters
-  used. And see below (``xvfb``) to understand when and why the ``DISPLAY`` environment variable
+  used. And see :ref:`xvfb` section to understand when and why the ``DISPLAY`` environment variable
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:136]:
```suggestion
  used. And see the :ref:`xvfb` section to understand when and why the ``DISPLAY`` environment variable
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 6]
```diff
@@ -174,39 +173,94 @@ QGIS Server is now available at http://localhost/qgisserver. To check, type in a
 
 .. index:: nginx, spawn-fcgi, fcgiwrap
 
-NGINX
-.....
+NGINX HTTP Server
+-----------------
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:177]:
```suggestion
.................
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 7]
```diff
@@ -174,39 +173,94 @@ QGIS Server is now available at http://localhost/qgisserver. To check, type in a
...
 Install NGINX:
 
 .. code-block:: bash
 
  apt install nginx
 
+
+* As a first option, you can use **spawn-fcgi** or **fcgiwrap** to start and manage the QGIS Server
+  processes. Official Debian packages exist for both. When you have no X server running and you need, for example,
+  printing,	you can use :ref:`xvfb <xvfb>`.
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:194]:
```suggestion
* As a first option, you can use **spawn-fcgi** or **fcgiwrap** to start and manage the
  QGIS Server processes.
  Official Debian packages exist for both.
  When you have no X server running and you need, for example,
  printing,	you can use :ref:`xvfb <xvfb>`.
```
Not a big deal, but one line was a bit long, so I adjusted the whole paragraph.

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 8]
```diff
@@ -174,39 +173,94 @@ QGIS Server is now available at http://localhost/qgisserver. To check, type in a
...
+* Another option is to rely on **Systemd**, the init system for GNU/Linux that most
+  Linux distributions use today.
+  One of the advantages of this method is that it requires no other components or
+  processes.
+  It’s meant to be simple, yet robust and efficient for production deployments.
+
+NGINX Configuration
+...................
+
+The **include fastcgi_params;** used in previous configuration is important
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:205]:
```suggestion
The **include fastcgi_params;** used in the previous configuration is important,
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 9]
```diff
@@ -174,39 +173,94 @@ QGIS Server is now available at http://localhost/qgisserver. To check, type in a
...
+ fastcgi_param  SERVER_ADDR        $server_addr;
+ fastcgi_param  SERVER_PORT        $server_port;
+ fastcgi_param  SERVER_NAME        $server_name;
+
+ # PHP only, required if PHP was built with --enable-force-cgi-redirect
+ fastcgi_param  REDIRECT_STATUS    200;
+
+Moreover, you can use some :ref:`qgis-server-envvar` to configure QGIS Server.
+In the NGINX configuration file :file:`/etc/nginx/nginx.conf`, you have to use ``fastcgi_param`` instruction to define
+these variables as shown below:
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:237]:
```suggestion
0123456789012345678901234567890123456789012345678901234567890123456789
In the NGINX configuration file, :file:`/etc/nginx/nginx.conf`, you have to use
``fastcgi_param`` instruction to define these variables as shown below:
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 10]
```diff
@@ -174,39 +173,94 @@ QGIS Server is now available at http://localhost/qgisserver. To check, type in a
...
+         gzip           off;
+         include        fastcgi_params;
+         fastcgi_param  QGIS_DEBUG              1;
+         fastcgi_param  QGIS_SERVER_LOG_FILE    /var/log/qgis/qgisserver.log;
+         fastcgi_param  QGIS_SERVER_LOG_LEVEL   0;
+         fastcgi_pass   unix:/var/run/qgisserver.socket;
+     }
+
+Fcgi wrappers
+.............
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:251]:
```suggestion
FastCGI wrappers
^^^^^^^^^^^^^^^^
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 11]
```diff
@@ -174,39 +173,94 @@ QGIS Server is now available at http://localhost/qgisserver. To check, type in a
...
+
+.. warning::
+
+  **fcgiwrap** is easier to set up than **spawn-fcgi**, because it's already wrapped
+  in a Systemd service. But it also leads to a solution that is much slower
+  than using spawn-fcgi. With fcgiwrap a new QGIS Server process is created
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:257]:
```suggestion
  than using spawn-fcgi. With fcgiwrap, a new QGIS Server process is created
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 12]
```diff
@@ -232,7 +286,7 @@ And restart NGINX to take into account the new configuration:
 
 .. code-block:: bash
 
- service nginx restart
+ systemctl restart nginx
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:289]:
```suggestion
  systemctl restart nginx
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 13]
```diff
@@ -243,21 +297,49 @@ have to manually start QGIS Server in your terminal:
...
-    With the above command spawn-fcgi spawns only one QGIS Server process. To use more than one QGIS
-    Server process you can combine spawn-fcgi with the
-    `multiwatch <https://redmine.lighttpd.net/projects/multiwatch/wiki>`_ tool, which is also
-    packaged in Debian.
+    When using spawn-fcgi, you may directly define environment variables
+    before running the server. For example:
+    ``export QGIS_SERVER_LOG_FILE=/var/log/qgis/qgisserver.log``
+
+Of course, you can add an init script to start QGIS Server at boot time or whenever you want. For example with **systemd**,
+edit the file :file:`/etc/systemd/system/qgis-server.service` with this content:
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:309]:
```suggestion
Of course, you can add an init script to start QGIS Server at boot time or whenever you want.
For example with **systemd**, edit the file
:file:`/etc/systemd/system/qgis-server.service` with this content:
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 14]
```diff
@@ -415,23 +445,24 @@ configuration located in :file:`/etc/systemd/system/qgis-server@.service`:
...
+
+   QGIS_PROJECT_FILE=/etc/qgis/myproject.qgs
+   QGIS_SERVER_LOG_STDERR=1
+   QGIS_SERVER_LOG_LEVEL=3
+
 Now start socket service:
 
 .. code-block:: bash
 
  sudo systemctl start qgis-server@sockets.service
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:463]:
```suggestion
  sudo systemctl start qgis-server@sockets.service
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 15]
```diff
@@ -456,14 +487,16 @@ Now restart NGINX for the new configuration to be taken into account:
...
 
 Thanks to Oslandia for sharing `their tutorial <https://oslandia.com/en/2018/11/23/deploying-qgis-server-with-systemd/>`_. 
 
+.. _xvfb:
+
 Xvfb
 ----
-
-QGIS Server needs a running X Server to be fully usable, in particular for printing. 
+
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:498]:
```suggestion
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 16]
```diff
@@ -473,31 +506,19 @@ To install the package:
...
-parameter or directly use **xvfb-run**.
-
-With Apache you just add to your ``Fcgi`` configuration (see above):
-
-.. code-block:: apache
-
- FcgidInitialEnv DISPLAY       ":99"
-
-Create the service file:
+Create the service file :file:`/etc/systemd/system/xvfb.service` with this content:
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:509]:
```suggestion
Create the service file, :file:`/etc/systemd/system/xvfb.service`, with this content:
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 17]
```diff
@@ -507,33 +528,45 @@ Enable, start and check the status of the ``xvfb.service``:
...
 
-.. code-block:: bash
+With Apache
+...........
 
- systemctl restart apache2
+Then you can configure the **DISPLAY** parameter.
 
-With NGINX and spawn-fcgi using ``xvfb-run``:
+With Apache you just add to your ``Fcgi`` configuration (see above):
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:539]:
```suggestion
With Apache you just add to your *FastCGI* configuration (see above):
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 18]
```diff
@@ -507,33 +528,45 @@ Enable, start and check the status of the ``xvfb.service``:
...
 
-.. code-block:: bash
+.. code-block:: apache
+
+ FcgidInitialEnv DISPLAY       ":99"
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:543]:
```suggestion
  FcgidInitialEnv DISPLAY       ":99"
```

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 19]
```diff
@@ -507,33 +528,45 @@ Enable, start and check the status of the ``xvfb.service``:
...
-                              -G www-data -U www-data -n
 
-The other option is to start a virtual X server environment with a specific
-display number thanks to **Xvfb**:
+Now restart Apache for the new configuration to be taken into account:
 
 .. code-block:: bash
 
- /usr/bin/Xvfb :99 -screen 0 1024x768x24 -ac +extension GLX +render -noreset
+ systemctl restart apache2
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:550]:
```suggestion
  systemctl restart apache2
```

**Comment by benoitdm-oslandia** [comment]
> > I don't plan to backport this to current LTR
> 
> Why not? This should be relevant also for 3.10, or?

Yes, you are right, it could be relevant for this version too ! I was working on this PR with my mind following the [server doc splitting](https://github.com/qgis/QGIS-Documentation/pull/5930)!

I fix the PR description.

**Comment by benoitdm-oslandia** [comment]
@havatv After following yours suggestions, I remembered why I put the "Testing the installation" and the "Deploying QGIS server" section under bullets instead of headings: it reduces the whole subtree headings by 1. In the toc, this makes the "Testing the installation" section at the same visibility than the more interesting part: "Deploying QGIS server". 

This is why I prefer the previous toc:
```
  Installation on Debian-based systems
    Apache HTTP Server
    NGINX HTTP Server
      NGINX Configuration
      FastCGI wrappers
      Systemd
    Xvfb
      With Apache
      With NGINX
```
to this new one:
```
  Installation on Debian-based systems
    Testing the installation
    Deploying QGIS server
      Apache HTTP Server
      NGINX HTTP Server
      Xvfb
```

Do you mind, If I revert to the previous heading levels?

File: docs/user_manual/working_with_ogc/server/getting_started.rst [hunk 20]
```diff
@@ -6,7 +6,7 @@ Getting Started
 
    .. contents::
       :local:
-      :depth: 2
+      :depth: 3
```
Comment (havatv) [inline docs/user_manual/working_with_ogc/server/getting_started.rst:9]:
```suggestion
      :depth: 4
```
What about increasing the depth?

**Comment by havatv** [comment]
> Do you mind, If I revert to the previous heading levels?

My concern was using bullets for this. We should avoid that. I have suggested to increase the toc depth.

**Comment by havatv** [comment]
Looks good. Can you fix the conflicts, so that we can merge?

**Comment by benoitdm-oslandia** [comment]
hi @havatv ! I had merge with the master. It is ready for you to merge !
Thanks

**Comment by SrNetoChan** [comment]
@havatv 
> Why not? This should be relevant also for 3.10, or?

Although this is relevant for 3.10, I follow the same opinion of @rduivenvoorde that we should try not to change the LTR documentation structure too much unless it is absolutely necessary. With the current server split, backporting this changes, without backporting the split, is a bit of a nightmare...

**Comment by rduivenvoorde** [comment]
Yes, please be carefull, every (minor) string change in 3.10 makes Sphinx not find the translation anymore (as the strings are the msg keys), so only fix errors in 3.10 and keep the strings there as stable as possible (else we have to update the source strings of Transifex every time, and with that translators 'loose' their old translations).

**Comment by benoitdm-oslandia** [comment]
> @havatv
> 
> > Why not? This should be relevant also for 3.10, or?
> 
> Although this is relevant for 3.10, I follow the same opinion of @rduivenvoorde that we should try not to change the LTR documentation structure too much unless it is absolutely necessary. With the current server split, backporting this changes, without backporting the split, is a bit of a nightmare...

Ok! So, no backport!

**Pull request merged**